### PR TITLE
HDDS-761. Create S3 subcommand to run S3 related operations

### PR DESCRIPTION
**Pull request by vivekratnavel**
This patch is to move all S3 related subcommands from `ozone sh s3` to `ozone s3`.

ozone sh s3 getsecret -> **ozone s3 getsecret**
ozone sh bucket path `<bucketname>` -> **ozone s3 path `<bucketname>`**

https://issues.apache.org/jira/browse/HDDS-761

### CODE REVIEW

**Comment by vivekratnavel** [comment]
@bharatviswa504 @elek @arp7 @swagle @avijayanhwx @hanishakoneru

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 42 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 2 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 33 | Maven dependency ordering for branch |
| +1 | mvninstall | 1197 | trunk passed |
| +1 | compile | 989 | trunk passed |
| +1 | checkstyle | 222 | trunk passed |
| -1 | mvnsite | 60 | integration-test in trunk failed. |
| +1 | shadedclient | 765 | branch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-hdds/docs hadoop-ozone/integration-test |
| +1 | findbugs | 124 | trunk passed |
| +1 | javadoc | 150 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 29 | Maven dependency ordering for patch |
| -1 | mvninstall | 22 | integration-test in the patch failed. |
| +1 | compile | 924 | the patch passed |
| +1 | javac | 924 | the patch passed |
| +1 | checkstyle | 219 | the patch passed |
| -1 | mvnsite | 45 | integration-test in the patch failed. |
| +1 | shellcheck | 26 | There were no new shellcheck issues. |
| +1 | shelldocs | 34 | There were no new shelldocs issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 759 | patch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-hdds/docs hadoop-ozone/integration-test |
| +1 | findbugs | 140 | the patch passed |
| +1 | javadoc | 152 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 35 | docs in the patch passed. |
| +1 | unit | 54 | common in the patch passed. |
| -1 | unit | 43 | integration-test in the patch failed. |
| +1 | unit | 68 | ozone-manager in the patch passed. |
| +1 | asflicense | 50 | The patch does not generate ASF License warnings. |
| | | 6850 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-579/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/579 |
| Optional Tests |  dupname  asflicense  mvnsite  unit  shellcheck  shelldocs  compile  javac  javadoc  mvninstall  shadedclient  findbugs  checkstyle  |
| uname | Linux 25e87bb3f470 4.4.0-138-generic #164~14.04.1-Ubuntu SMP Fri Oct 5 08:56:16 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / c072458 |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| mvnsite | https://builds.apache.org/job/hadoop-multibranch/job/PR-579/1/artifact/out/branch-mvnsite-hadoop-ozone_integration-test.txt |
| shellcheck | v0.4.6 |
| findbugs | v3.1.0-RC1 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-579/1/artifact/out/patch-mvninstall-hadoop-ozone_integration-test.txt |
| mvnsite | https://builds.apache.org/job/hadoop-multibranch/job/PR-579/1/artifact/out/patch-mvnsite-hadoop-ozone_integration-test.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-579/1/artifact/out/patch-unit-hadoop-ozone_integration-test.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-579/1/testReport/ |
| Max. process+thread count | 306 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/docs hadoop-ozone/common hadoop-ozone/integration-test hadoop-ozone/ozone-manager U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-579/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 19 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 2 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 26 | Maven dependency ordering for branch |
| +1 | mvninstall | 1097 | trunk passed |
| +1 | compile | 1042 | trunk passed |
| +1 | checkstyle | 216 | trunk passed |
| +1 | mvnsite | 178 | trunk passed |
| +1 | shadedclient | 758 | branch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-hdds/docs hadoop-ozone/integration-test |
| +1 | findbugs | 127 | trunk passed |
| +1 | javadoc | 148 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 26 | Maven dependency ordering for patch |
| -1 | mvninstall | 22 | integration-test in the patch failed. |
| +1 | compile | 938 | the patch passed |
| +1 | javac | 938 | the patch passed |
| +1 | checkstyle | 256 | the patch passed |
| +1 | mvnsite | 199 | the patch passed |
| +1 | shellcheck | 26 | There were no new shellcheck issues. |
| +1 | shelldocs | 33 | There were no new shelldocs issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 722 | patch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-hdds/docs hadoop-ozone/integration-test |
| +1 | findbugs | 139 | the patch passed |
| +1 | javadoc | 144 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 32 | docs in the patch passed. |
| +1 | unit | 55 | common in the patch passed. |
| -1 | unit | 963 | integration-test in the patch failed. |
| -1 | unit | 56 | ozone-manager in the patch failed. |
| +1 | asflicense | 50 | The patch does not generate ASF License warnings. |
| | | 7672 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.om.TestScmChillMode |
|   | hadoop.ozone.container.server.TestSecureContainerServer |
|   | hadoop.ozone.om.TestOzoneManagerHttpServer |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-579/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/579 |
| Optional Tests |  dupname  asflicense  mvnsite  unit  shellcheck  shelldocs  compile  javac  javadoc  mvninstall  shadedclient  findbugs  checkstyle  |
| uname | Linux dc5dd806f6e4 4.4.0-139-generic #165~14.04.1-Ubuntu SMP Wed Oct 31 10:55:11 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 599e926 |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| shellcheck | v0.4.6 |
| findbugs | v3.1.0-RC1 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-579/2/artifact/out/patch-mvninstall-hadoop-ozone_integration-test.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-579/2/artifact/out/patch-unit-hadoop-ozone_integration-test.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-579/2/artifact/out/patch-unit-hadoop-ozone_ozone-manager.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-579/2/testReport/ |
| Max. process+thread count | 3788 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/docs hadoop-ozone/common hadoop-ozone/integration-test hadoop-ozone/ozone-manager U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-579/2/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 0 | Docker mode activated. |
| -1 | patch | 6 | https://github.com/apache/hadoop/pull/579 does not apply to trunk. Rebase required? Wrong Branch? See https://wiki.apache.org/hadoop/HowToContribute for help. |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hadoop/pull/579 |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-579/4/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 0 | Docker mode activated. |
| -1 | patch | 6 | https://github.com/apache/hadoop/pull/579 does not apply to trunk. Rebase required? Wrong Branch? See https://wiki.apache.org/hadoop/HowToContribute for help. |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hadoop/pull/579 |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-579/5/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 0 | Docker mode activated. |
| -1 | patch | 9 | https://github.com/apache/hadoop/pull/579 does not apply to trunk. Rebase required? Wrong Branch? See https://wiki.apache.org/hadoop/HowToContribute for help. |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hadoop/pull/579 |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-579/6/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/ozShell/TestS3Shell.java [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+
+  private final ByteArrayOutputStream out = new ByteArrayOutputStream();
+  private final ByteArrayOutputStream err = new ByteArrayOutputStream();
+  private static final PrintStream OLD_OUT = System.out;
+  private static final PrintStream OLD_ERR = System.err;
+
+  @Parameterized.Parameters
+  public static Collection<Object[]> clientProtocol() {
+    Object[][] params = new Object[][] {
+        {RpcClient.class}};
```
Comment (bharatviswa504) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/ozShell/TestS3Shell.java]: We don't need parameterized tests, as for S3 we don't support RestClient.
And also it is having only one param RpcClient.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/ozShell/TestS3Shell.java [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+    return Arrays.asList(params);
+  }
+
+  @Parameterized.Parameter
+  @SuppressWarnings("visibilitymodifier")
```
Comment (bharatviswa504) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/ozShell/TestS3Shell.java]: Why do we need this?

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 0 | Docker mode activated. |
| -1 | patch | 5 | https://github.com/apache/hadoop/pull/579 does not apply to trunk. Rebase required? Wrong Branch? See https://wiki.apache.org/hadoop/HowToContribute for help. |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hadoop/pull/579 |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-579/7/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by vivekratnavel** [comment]
Failures are unrelated to this patch

**Comment by bharatviswa504** [comment]
+1 LGTM.
Pending jenkins

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 40 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 2 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 63 | Maven dependency ordering for branch |
| +1 | mvninstall | 1082 | trunk passed |
| +1 | compile | 967 | trunk passed |
| +1 | checkstyle | 219 | trunk passed |
| -1 | mvnsite | 46 | integration-test in trunk failed. |
| +1 | shadedclient | 774 | branch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-hdds/docs hadoop-ozone/dist hadoop-ozone/integration-test |
| +1 | findbugs | 127 | trunk passed |
| +1 | javadoc | 182 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 26 | Maven dependency ordering for patch |
| -1 | mvninstall | 19 | dist in the patch failed. |
| -1 | mvninstall | 25 | integration-test in the patch failed. |
| +1 | compile | 916 | the patch passed |
| +1 | javac | 916 | the patch passed |
| +1 | checkstyle | 220 | the patch passed |
| +1 | mvnsite | 209 | the patch passed |
| +1 | shellcheck | 25 | There were no new shellcheck issues. |
| +1 | shelldocs | 33 | There were no new shelldocs issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 752 | patch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-hdds/docs hadoop-ozone/dist hadoop-ozone/integration-test |
| +1 | findbugs | 142 | the patch passed |
| +1 | javadoc | 182 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 35 | docs in the patch passed. |
| +1 | unit | 54 | common in the patch passed. |
| +1 | unit | 36 | dist in the patch passed. |
| -1 | unit | 926 | integration-test in the patch failed. |
| +1 | unit | 66 | ozone-manager in the patch passed. |
| +1 | asflicense | 54 | The patch does not generate ASF License warnings. |
| | | 7830 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.om.TestScmChillMode |
|   | hadoop.ozone.TestStorageContainerManager |
|   | hadoop.ozone.container.TestContainerReplication |
|   | hadoop.ozone.client.rpc.TestFailureHandlingByClient |
|   | hadoop.ozone.client.rpc.TestSecureOzoneRpcClient |
|   | hadoop.ozone.om.TestOzoneManager |
|   | hadoop.ozone.scm.node.TestSCMNodeMetrics |
|   | hadoop.ozone.client.rpc.TestBCSID |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-579/8/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/579 |
| Optional Tests |  dupname  asflicense  mvnsite  unit  shellcheck  shelldocs  compile  javac  javadoc  mvninstall  shadedclient  findbugs  checkstyle  |
| uname | Linux 8244fe72a8da 4.4.0-138-generic #164~14.04.1-Ubuntu SMP Fri Oct 5 08:56:16 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 2627dad |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| mvnsite | https://builds.apache.org/job/hadoop-multibranch/job/PR-579/8/artifact/out/branch-mvnsite-hadoop-ozone_integration-test.txt |
| shellcheck | v0.4.6 |
| findbugs | v3.1.0-RC1 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-579/8/artifact/out/patch-mvninstall-hadoop-ozone_dist.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-579/8/artifact/out/patch-mvninstall-hadoop-ozone_integration-test.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-579/8/artifact/out/patch-unit-hadoop-ozone_integration-test.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-579/8/testReport/ |
| Max. process+thread count | 4455 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/docs hadoop-ozone/common hadoop-ozone/dist hadoop-ozone/integration-test hadoop-ozone/ozone-manager U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-579/8/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Pull request merged**